raft.go

```diff
@@ -1770,7 +1778,11 @@ func (r *raft) handleAppendEntries(m pb.Message) {
 }
 
 func (r *raft) handleHeartbeat(m pb.Message) {
-	r.raftLog.commitTo(m.Commit)
+	// If our log is not a prefix of the leader's log, it is unsafe to advance the
+	// commit index, because the entries at this index may mismatch.
+	if r.logSynced {
```

Comment:
BTW, `logSynced == true` iff `r.raftLog.lastTerm() == r.term`. I'll add a comment to this extend.

But also, we can remove the `r.logSynced` field and do the term check instead, if that's preferrable.

Reply:
Done.